pkg/services/ngalert/notifier/images.go

```diff
@@ -20,21 +20,27 @@ func newImageStore(store store.ImageStore) images.ImageStore {
 	}
 }
 
-func (i imageStore) GetImage(ctx context.Context, token string) (*images.Image, error) {
-	image, err := i.store.GetImage(ctx, token)
-	if err != nil {
-		if errors.Is(err, models.ErrImageNotFound) {
-			err = images.ErrImageNotFound
-		}
+func (i imageStore) GetImage(ctx context.Context, uri string) (*images.Image, error) {
+	var (
+		image *models.Image
+		err   error
+	)
+
+	// Check whether the uri is a URL or a token to know how to query the DB.
+	if strings.HasPrefix(uri, "http") {
```

Comment:
I would have done the inverse and check for `token://`? You can assume all other inputs are a URL

Reply:
I wanted to cover the case in which the records in the database correspond to previous versions of Grafana, so no `token://` prefix.
Checking for `http` also checks for `https`.

Reply:
the plain token is possible right after the start when alertmanager has just started and loaded the state from disk.